### ISSUE TITLE
feature request: Local tagmanager in head of tagpill dropdown

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#1112]
Please first look at this image:

![localtagmanager](https://cloud.githubusercontent.com/assets/174761/5154857/1e0d4a9c-7268-11e4-9a24-69429e749339.png)

Chances are you're looking at the tagpill when you realize you want to manage it. This setup would give immediate access via a discrete "localtagmanager dropdown arrow". In the image, I have used the "info slider" layout. But I cheated a little because I don't know if the tag menu can stay that slim if the dropdown icon-name list contains wider titles. (Would it make the original dropdown expanded already before the local tagmanager.was opened then?)

An alternative could be a horizontal slider and, I guess, then a right pointing arrow instead.

One thing omitted from the current tag manager is the colored box that gives a palette. Maybe I should have included it but I figure you see the resulting color on the actual tagpill so there is little point in repeating the color blob twice visually. I put a small arrow next to the word "Color" instead as an optional place for a popup palette.

Also, if the changes get reflected immediately ("live") in the actual tagpill then it is not necessary to include the display inside the local tagmanager.

Obviously not an urgent matter but still "would be nice if..."

<:-)

**Comment by tobibeer** [comment]
> autocollapse

not really a fan of this ...so long as clicking that title doesn't also open the info-panel

**Comment by pmario** [comment]
@twMat ... In the light of your [TagConfig plugin](https://talk.tiddlywiki.org/t/presenting-tagconfig-customize-tags-directly-in-tiddler-edit-mode/12993). Do you want to keep this one open?

**Comment by twMat** [comment]
@pmario - right, thanks.

I do still think that *something* like the OP should be in the core. But what is *really* needed, not least for plugin makers, is... is it called "hooks"?.... something that allows plugin makers to "insert" their solution without having to overwrite any core shadow tids. (In my later plugins, I've started to semi circumvent this by actively demanding that the user manually injects stuff into the core tids via a checkbox, but that is still not optimal.) I'll start a separate discussion on this... but not at this moment...

**Comment by pmario** [comment]
You are right. Since the current edit template has no clickable area in the tag-pill, it should be straight forward to implement procedure, that can be overwritten. 

The element to be changed in the template is `<$view field="title" format="text"/>` -- May be into
`<$transclude $variable="tag-body-inner-title"/>` ... which then could contain your code. 

Just an idea. Not tested. You could use this text and **create a new issue**, which I think should be actionable, if Jeremy is OK with it. 

```
			<$transclude tiddler=<<icon>>/>
			<$view field="title" format="text"/>
			<$button class="tc-btn-invisible tc-remove-tag-button"
				style.fill=<<foregroundColor>>
			>
```

**Comment by twMat** [comment]
@pmario 

> ... implement procedure, that can be overwritten.

To avoid overwriting anything from the core, could it be an *empty* procedure call somehow? E.g `<<tp-edit-view-tag-pill>>` or some such?

**Comment by pmario** [comment]
We have to replace the existing view-widget anyhow, otherwise we will see 2 tag-titles. So IMO overwriting 1 line of code should not put any extra maintenance burden on your plugin. And it will be the simplest possibility.
